openarm/bus/__init__.py

```diff
@@ -28,46 +75,71 @@ def send(self, msg: can.Message, timeout: float | None = None) -> None:
         """
         self.bus.send(msg, timeout)
 
-    def recv(
-        self, arbitration_id: int, timeout: float | None = None
-    ) -> can.Message | None:
+    def recv(self, arbitration_id: int) -> Coroutine[any, any, can.Message]:
         """Receive a CAN message with the specified arbitration ID.
 
-        Messages with other arbitration IDs are queued for later retrieval.
+        Messages with other arbitration IDs are queued separately.
 
         Args:
             arbitration_id: The arbitration ID to filter for.
-            timeout: Optional receive timeout in seconds. None means wait indefinitely.
 
         Returns:
-            The received CAN message, or None if timeout occurred.
+            The received CAN message.
+
+        """
+        # defaultdict automatically creates queue if it doesn't exist
+        return self.queues[arbitration_id].get()
+
+
+class AsyncBus:
+    """Async CAN bus wrapper that receives all messages."""
+
+    def __init__(self, bus: can.BusABC) -> None:
+        """Initialize the broadcast bus.
+
+        Args:
+            bus: The underlying CAN bus interface.
+
+        """
+        self.bus = bus
+        self.queue: Queue[can.Message] = Queue()
```

Comment:
I think this should be private:
```suggestion
        self._queue: Queue[can.Message] = Queue()
```

Reply:
done